cursive-core/src/utils/markup/markdown.rs

```diff
@@ -76,44 +112,117 @@ impl Iterator for Parser<'_> {
     fn next(&mut self) -> Option<Self::Item> {
         loop {
             let next = self.parser.next()?;
-
             match next {
                 Event::Start(tag) => match tag {
                     // Add to the stack!
                     Tag::Emphasis => self.stack.push(Style::from(Effect::Italic)),
                     Tag::Heading { level, .. } => {
-                        let lines = if self.first { "" } else { "\n\n" };
+                        /* Force a new paragraph before headings, but only if this is not
+                         * the first output, and we have not skipped lines already.
+                         */
+                        let lines = if self.first {
+                            ""
+                        } else if !self.new_paragraph {
+                            self.new_paragraph = true;
+                            "\n\n"
+                        } else {
+                            self.new_paragraph = false;
+                            ""
+                        };
+                        /* Headings are bold by default */
+                        self.stack.push(Style::from(Effect::Bold));
                         return Some(self.literal(format!(
                             "{}{} ",
                             lines,
                             heading(level as usize)
                         )));
                     }
-                    Tag::BlockQuote(_) => return Some(self.literal("> ")),
+                    Tag::BlockQuote(_) => {
+                        /* Push italic style */
+                        self.stack.push(Style::from(Effect::Italic));
+                        /* Force a blank line before a blockquote */
+                        self.new_paragraph = true;
+                        return Some(self.literal("\n\n> "));
```

Comment:
Should we still insert 2 newlines, even if this is the first thing in the document?